Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -65,16 +68,23 @@ sub initPlugin {
 		before => 'top',
 		func   => \&trackInfoMenu,
 	) );
+	
+	%recentlyPlayed = map { $_->{url} => $_ } reverse @{$prefs->get('recent')};
 
 	$class->SUPER::initPlugin(
 		feed   => \&handleFeed,
 		tag    => 'podcasts',
 		menu   => 'apps',
 	);
-	
+		
 	$class->addNonSNApp();
 }
 
+sub shutdownPlugin {
+	my @played = values %recentlyPlayed;
```

Comment:
Is the order, in which `Tie::Cache::LRU` would return values defined?

Reply:
the idea is that we load them in reverse order in init() to neutralize changes made here

Reply:
Huh? If `values` returned the values in random order (I'm questioning this, not saying that's the case), then `reverse` wouldn't be of any help at all, right? It would only, well, reverse, whatever random order we got. As `Tie::Cache::LRU` seems to be based on an array, I could imagine the order is well defined. But then we're going through a hash implementation...

If you tested this, then that's good. If not, it might be worth some testing.

Reply:
Wrote this little test script:

```perl
#!/usr/bin/perl

use Tie::Cache::LRU

tie my %lruList, 'Tie::Cache::LRU', 256;
%lruList = ();

foreach (qw(one two three four five six seven eight nine ten)) {
	$lruList{$_} = $_;
}

print join(', ', values %lruList, "\n");

print $lruList{one} . "\n";

print join(', ', values %lruList, "\n");

$lruList{one} = 1;

print join(', ', values %lruList, "\n");

1;
```

Which shows that not only would the list of returned values _not_ be sorted by their LRU state, but also the items are returned in random order on every run:

```
$ ./lru.pl 
one, three, two, four, eight, five, six, seven, nine, ten, 
one
one, three, two, four, eight, five, six, seven, nine, ten, 
1, three, two, four, eight, five, six, seven, nine, ten, 

$ ./lru.pl 
nine, six, seven, two, three, ten, one, four, five, eight, 
one
nine, six, seven, two, three, ten, one, four, five, eight, 
nine, six, seven, two, three, ten, 1, four, five, eight, 

$ ./lru.pl 
three, eight, nine, six, four, five, ten, one, two, seven, 
one
three, eight, nine, six, four, five, ten, one, two, seven, 
three, eight, nine, six, four, five, ten, 1, two, seven, 

$ ./lru.pl 
eight, nine, six, seven, four, three, ten, five, two, one, 
one
eight, nine, six, seven, four, three, ten, five, two, one, 
eight, nine, six, seven, four, three, ten, five, two, 1, 

$ ./lru.pl 
one, four, six, eight, five, nine, ten, seven, two, three, 
one
one, four, six, eight, five, nine, ten, seven, two, three, 
1, four, six, eight, five, nine, ten, seven, two, three, 
```

Therefore I believe you'll have to handle the sorting order of that list yourself if you want to keep a consistent order.

Reply:
I will verify again, but in all the tests so far it works and this is also the same thing being used in YT plugin and 2 others

Reply:
oh, the declaration must be before the tie, otherwise it's just a hash

Reply:
> I will verify again, but in all the tests so far it works and this is also the same thing being used in YT plugin and 2 others

Ok, that's reassuring :-). What am I missing?

> oh, the declaration must be before the tie, otherwise it's just a hash

Heh!... lesson learned! And I took my code straight out of LMS...